presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -168,19 +169,28 @@ else if (config.getPinGlueClientToCurrentRegion()) {
             }
         }
 
+        AWSCredentialsProvider credentialsProvider = createAwsCredentialsProvider(config);
+        asyncGlueClientBuilder.setCredentials(credentialsProvider);
+        return asyncGlueClientBuilder.build();
+    }
+
+    private static AWSCredentialsProvider createAwsCredentialsProvider(GlueHiveMetastoreConfig config)
+    {
+        AWSCredentialsProvider credentialsProvider;
```

Comment:
There is no need for this variable. Just `return` in each `if` block. Then, don't use `else` (will become redundant).

Reply:
In general, I like reducing the independent paths (cyclomatic complexity), but don't mind making the change.